regression/pages/studio/studio_home.py

```diff
@@ -31,15 +31,11 @@ def select_course(self, course_title):
         """
         Selects the course we want to perform tests on
         """
-        course_title_selector = '.course-link h3'
-        self.wait_for_element_presence(course_title_selector,
-                                       'Course link presence')
-        course_names = self.q(css=course_title_selector)
-        for vals in course_names:
-            if course_title in vals.text:
-                vals.click()
-                return
-        raise BrokenPromise('Course title not found')
+        query = self.q(xpath='//h3[contains(text(), "{}")]'.format(course_title)).first
+        if course_title in query.text:
```

Comment:
Why wouldn't this find the course title, given that that's what your Xpath searched for?

Reply:
It would, but there needs to be some way to verify that the element was found, otherwise the BrokenPromise will never be raised. The `click()` method doesn’t return anything, so a one-liner wouldn’t be enough.